src/installer/tests/HostActivation.Tests/MultiArchInstallLocation.cs

```diff
@@ -71,6 +71,49 @@ public void EnvironmentVariable_ArchSpecificDotnetRootIsUsedOverDotnetRoot()
                 .And.NotHaveStdErrContaining("Using environment variable DOTNET_ROOT=");
         }
 
+        [Fact]
+        public void EnvironmentVariable_DotnetRootPathDoesNotExist()
+        {
+            var fixture = sharedTestState.PortableAppFixture
+                .Copy();
+
+            var appExe = fixture.TestProject.AppExe;
+            Command.Create(appExe)
+                .EnableTracingAndCaptureOutputs()
+                .DotNetRoot("non_existent_path")
+                .EnvironmentVariable("DOTNET_MULTILEVEL_LOOKUP", "0")
+                .EnvironmentVariable(
+                    Constants.TestOnlyEnvironmentVariables.DefaultInstallPath,
+                    sharedTestState.InstallLocation)
```

Comment:
You'll also need to configure the app to actually check the test-only environment variable - there's the `TestOnlyProductBehavior` class, which is also used by `RegisteredInstallLocationOverride`.